centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInputs/hooks/useDatasetFilter.ts

```diff
@@ -0,0 +1,294 @@
+};
+
+const resourceQueryParameters = [
+  {
+    name: 'limit',
+    value: 30
+  }
+];
+
+const useDatasetFilter = (
+  datasetFilter: Array<Dataset>,
+  datasetFilterIndex: number
+): UseDatasetFilterState => {
+  const { values, setFieldValue, setFieldTouched, touched } =
+    useFormikContext<FormikValues>();
+
+  const value = useMemo<Array<Dataset> | undefined>(
+    () =>
+      path<Array<Dataset> | undefined>(
+        ['data', 'datasetFilters', datasetFilterIndex],
+        values
+      ),
+    [
+      path<Array<Dataset> | undefined>(
+        ['data', 'datasetFilters', datasetFilterIndex],
+        values
+      )
+    ]
+  );
+
+  const lowestResourceTypeReached = (): boolean =>
+    equals(last(datasetFilter)?.resourceType, ResourceTypeEnum.Service) ||
+    equals(last(datasetFilter)?.resourceType, ResourceTypeEnum.MetaService);
+
+  const getResourceTypeOptions = (index: number): Array<SelectEntry> => {
+    if (isNil(value)) {
+      return resourceTypeOptions;
+    }
+
+    if (equals(value[index - 1]?.resourceType, ResourceTypeEnum.HostCategory)) {
+      return [
+        { id: ResourceTypeEnum.Host, name: labelHost },
+        { id: ResourceTypeEnum.HostGroup, name: labelHostGroup },
+        { id: ResourceTypeEnum.Service, name: labelService },
+        { id: ResourceTypeEnum.ServiceCategory, name: labelServiceCategory },
+        { id: ResourceTypeEnum.ServiceGroup, name: labelServiceGroup }
+      ];
+    }
+
+    if (equals(value[index - 1]?.resourceType, ResourceTypeEnum.HostGroup)) {
+      return [
+        { id: ResourceTypeEnum.Host, name: labelHost },
+        { id: ResourceTypeEnum.HostCategory, name: labelHostCategory },
+        { id: ResourceTypeEnum.Service, name: labelService },
+        { id: ResourceTypeEnum.ServiceCategory, name: labelServiceCategory },
+        { id: ResourceTypeEnum.ServiceGroup, name: labelServiceGroup }
+      ];
+    }
+
+    if (equals(value[index - 1]?.resourceType, ResourceTypeEnum.Host)) {
+      return [
+        { id: ResourceTypeEnum.Service, name: labelService },
+        { id: ResourceTypeEnum.ServiceCategory, name: labelServiceCategory },
+        { id: ResourceTypeEnum.ServiceGroup, name: labelServiceGroup }
+      ];
+    }
+
+    if (
+      equals(value[index - 1]?.resourceType, ResourceTypeEnum.ServiceCategory)
+    ) {
+      return [
+        { id: ResourceTypeEnum.Service, name: labelService },
+        { id: ResourceTypeEnum.ServiceGroup, name: labelServiceGroup }
+      ];
+    }
+
+    if (equals(value[index - 1]?.resourceType, ResourceTypeEnum.ServiceGroup)) {
+      return [
+        { id: ResourceTypeEnum.Service, name: labelService },
+        { id: ResourceTypeEnum.ServiceCategory, name: labelServiceCategory }
+      ];
+    }
+
+    if (
+      equals(value[index - 1]?.resourceType, ResourceTypeEnum.MetaService) ||
+      equals(value[index - 1]?.resourceType, ResourceTypeEnum.Service)
+    ) {
+      return [];
+    }
```

Comment:
you can also simplify this. check what is done [here ](https://github.com/centreon/centreon/blob/develop/centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/useResources.ts#L198)

Reply:
done